aggregation: Add dimensions to transaction metrics (#7033)
elastic/apm-server: docs/legacy/metricset-indices.asciidoc

* `host.os.platform`: The platform name of the service that served the transaction, for example `linux`.
* `container.id`: The container ID of the service that served the transaction
* `kubernetes.pod.name`: The name of the Kubernetes pod running the service that served the transaction
* `cloud.provider`: The cloud provider hosting the service instance that served the transaction
* `cloud.region`: The cloud region hosting the service instance that served the transaction
* `cloud.availability_zone`: The cloud availability zone hosting the service instance that served the transaction
* `cloud.account.id`: The cloud account id of the service that served the transaction
* `cloud.account.name`: The cloud account name of the service that served the transaction
* `cloud.machine.type`: The cloud machine type or instance type of the service that served the transaction
* `cloud.project.id`: The cloud project identifier of the service that served the transaction
* `cloud.project.name`: The cloud project name of the service that served the transaction
* `cloud.service.name`: The cloud service name of the service that served the transaction

> * `faas.coldstart`: Whether a service running  of the service that served the transaction

Author: ```suggestion
* `faas.coldstart`: Whether the _serverless_ service that served the transaction had a cold start.
```
